### PR TITLE
Add General/Routing, Issue #281

### PR DESCRIPTION
**Pull request by SSwiniarski**
### Description

Add entry for General/Routing.

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.

### CODE REVIEW

**Comment by Dusch4593** [comment]
I'll take on the first review for this one 😄

File: content/general/concepts/routing/routing.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+When one device communicates to another across a network, such as a PC communicating to a printer across a LAN, or a phone's web browser communicating to a web server over the Internet, there are generally multiple ways the signals, called packets, can travel from the source to the destination. The process of deciding what path packets travel across a network is called routing. The process of routing is executed by devices called routers that live on the network where there are junctions (nodes) between these various paths.
+
+A router can be physical piece of hardware, or it can be software based. Software-based routers tend to be more limited in functionality.
+
+In deciding paths to route traffic, a router always has some default path to send traffic with no specific destination specified. For traffic with a specific destination, if there is more than one path, the router can make a decision on what path to use based on several criteria, which can include:
+
+- Hop Count
+- Bandwidth
+- Delay/Latency
+
```
Comment (Dusch4593) [inline content/general/concepts/routing/routing.md:25]:
```suggestion

## Type of Routing

```

File: content/general/concepts/routing/routing.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  - 'Web Development'
+Tags:
+  - 'Cybersecurity'
+  - 'Networking'
+CatalogContent:
+  - 'learn/introduction-to-cybersecurity'
+  - 'paths/computer-science'
+---
+
+When one device communicates to another across a network, such as a PC communicating to a printer across a LAN, or a phone's web browser communicating to a web server over the Internet, there are generally multiple ways the signals, called packets, can travel from the source to the destination. The process of deciding what path packets travel across a network is called routing. The process of routing is executed by devices called routers that live on the network where there are junctions (nodes) between these various paths.
```
Comment (Dusch4593) [inline content/general/concepts/routing/routing.md:16]:
```suggestion
Routing is the process of deciding what paths communication signals may take across one or more networks. This process is used in many scenarios related to computer networking, including:

- A PC communicating with a printer across a Local Area Network, or LAN.
- A phone's web browser communicating with a web server over the internet.


Through routing, these communication signals, called packets, can take multiple paths from the source to the destination.
```

File: content/general/concepts/routing/routing.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+A router can be physical piece of hardware, or it can be software based. Software-based routers tend to be more limited in functionality.
```
Comment (Dusch4593) [inline content/general/concepts/routing/routing.md:18]:
```suggestion
## Routers

The process of routing is executed by devices known as  routers. They live on the network where there are junctions (nodes) between these various paths. Routers can either be physical pieces of hardware or software based, which tend to be more limited in functionality.
```

File: content/general/concepts/routing/routing.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
+---
+Title: 'Routing'
+Description: 'The process of deciding what path packets travel across a network is called routing.'
```
Comment (Dusch4593) [inline content/general/concepts/routing/routing.md:3]:
```suggestion
Description: 'Routing is the process of deciding what paths communication signals may take across one or more networks.'
```

File: content/general/concepts/routing/routing.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+When one device communicates to another across a network, such as a PC communicating to a printer across a LAN, or a phone's web browser communicating to a web server over the Internet, there are generally multiple ways the signals, called packets, can travel from the source to the destination. The process of deciding what path packets travel across a network is called routing. The process of routing is executed by devices called routers that live on the network where there are junctions (nodes) between these various paths.
+
+A router can be physical piece of hardware, or it can be software based. Software-based routers tend to be more limited in functionality.
+
+In deciding paths to route traffic, a router always has some default path to send traffic with no specific destination specified. For traffic with a specific destination, if there is more than one path, the router can make a decision on what path to use based on several criteria, which can include:
+
+- Hop Count
+- Bandwidth
+- Delay/Latency
```
Comment (Dusch4593) [inline content/general/concepts/routing/routing.md:24]: Could be expand on what each of these items are in a full sentence?

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@SSwiniarski Just finished my review! Let us know when you've finished looking over the comments/suggestions.

**Comment by SSwiniarski** [comment]
@Dusch4593, I implemented all your suggestions.

**Review by Dusch4593 — APPROVED** [review]
Looks good to move on to review 2️⃣!

File: content/general/concepts/routing/routing.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+- A PC communicating with a printer across a Local Area Network, or LAN.
+- A phone's web browser communicating with a web server over the internet.
+
+Through routing, these communication signals, called packets, can take multiple paths from the source to the destination.
+
+## Routers
+
+The process of routing is executed by devices known as routers. They live on the network where there are junctions (nodes) between these various paths. Routers can either be physical pieces of hardware or software based, which tend to be more limited in functionality.
+
+In deciding paths to route traffic, a router always has some default path to send traffic with no specific destination specified. For traffic with a specific destination, if there is more than one path, the router can make a decision on what path to use based on several criteria, which can include:
```
Comment (yangc95) [inline content/general/concepts/routing/routing.md:27]: Can we be explicit about what kind of traffic we're referring to here?

File: content/general/concepts/routing/routing.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+- Delay/Latency: The time between a signal sent on a route, and the response received.
+
+## Type of Routing
+
+There are several types of routing, based on the characteristics of the destination:
+
+- Unicast Routing: the packet is being sent to a single specified location.
+- Broadcast Routing: the packet is sent to all nodes on the network, even nodes that haven't requested it.
+- Multicast Routing: like broadcast routing, but packets are only sent to nodes that want them.
+- Anycast Routing: the destination consists of several devices with the same logical address, and the packet is routed to the closest of these devices.
```
Comment (yangc95) [inline content/general/concepts/routing/routing.md:40]:
```suggestion
- Unicast Routing: The packet is being sent to a single specified location.
- Broadcast Routing: The packet is sent to all nodes on the network, even nodes that haven't requested it.
- Multicast Routing: Similar to broadcast routing, but packets are only sent to nodes that want them.
- Anycast Routing: The destination consists of several devices with the same logical address and the packet is routed to the closest of these devices.
```

**Review by yangc95 — COMMENTED** [review]
Solid 👍  Review 2️⃣ done.

File: content/general/concepts/routing/routing.md (hunk 6 above)
Comment (yangc95) [inline content/general/concepts/routing/routing.md:27]:
```suggestion
A router always has some default path to send network traffic if no specific destination is specified. For traffic with a specific destination, if there is more than one path, the router can make a decision on what path to use based on several criteria, which can include:
```

**Comment by SSwiniarski** [comment]
@yangc95, @Dusch4593, I've updated this PR with the latest suggestions.

**Review by yangc95 — APPROVED** [review]
Looks good to me!

**Comment by Dusch4593** [comment]
Looks good to merge! 👍 I made a tiny edit in 5a793ba14baf4811bf184d4c7dea57106aa2a2c1

cc: @SSwiniarski @yangc95

**Comment by Dusch4593** [comment]
Nice! Live changes here:

https://github.com/Codecademy/docs/pull/502

**Pull request merged**